master/internal/db/postgres_experiments.go

```diff
@@ -937,18 +937,20 @@ func ActiveLogPolicies(
 // ExperimentTotalStepTime returns the total elapsed time for all allocations of the experiment
 // with the given ID. Any step with a NULL end_time does not contribute. Elapsed time is
 // expressed as a floating point number of seconds.
-func (db *PgDB) ExperimentTotalStepTime(id int) (float64, error) {
+func ExperimentTotalStepTime(ctx context.Context, id int) (float64, error) {
 	var seconds float64
-	if err := db.sql.Get(&seconds, `
-SELECT COALESCE(extract(epoch from sum(a.end_time - a.start_time)), 0)
-FROM allocations a
-JOIN run_id_task_id tasks ON a.task_id = tasks.task_id
-JOIN trials t ON tasks.run_id = t.id
-WHERE t.experiment_id = $1
-`, id); err != nil {
-		return 0, errors.Wrapf(err, "querying for total step time of experiment %v", id)
-	}
-	return seconds, nil
+	if err := Bun().NewSelect().
+		ColumnExpr("COALESCE(extract(epoch from sum(a.end_time - a.start_time)), 0)").
+		TableExpr("allocations AS a").
+		Join("JOIN run_id_task_id AS tasks").
+		JoinOn("a.task_id = tasks.task_id").
+		Join("JOIN trials AS t").
+		JoinOn("tasks.run_id = t.id").
+		Where("t.experiment_id = ?", id).
+		Scan(ctx, &seconds); err != nil {
+		return 0.0, errors.Wrapf(err, "querying for total step time of experiment %v", id)
+	}
+	return float64(seconds), nil
```

Comment:
I think the linter is going to complain about this cast

Have you run ```make -C master check```?

Reply:
Noticed that I would not need `float64(seconds)` type cast as I have already declared the variable in the start of the function, so removing it. Hence `make -C master check` works successfully!